### PR TITLE
added dnd-textarea webcomponent

### LINKED ISSUE #1704: Feature: Drag and drop for PSBT import
Background for this: Could you also create a template for this textarea (+the javascript code that makes the `ondrop`, ... work)?    Then the `psbt-uploader`  could have the same drag&drop effects.

_Originally posted by @relativisticelectron in https://github.com/cryptoadvance/specter-desktop/pull/1682#discussion_r860205314_

Suggested approach:
- Make drag and drop web component analogously to the new file-uploader web component. You can take the main JS code from the code mentionened above
- Apply this new web component to the PSBT import and enable drag and drop functionality here as well:

image

### PR DESCRIPTION
**Pull request by b30wulffz**
fixes #1704

Implemented a textarea with drag and drop functionality.
Upon dragging and dropping the file in the textarea, the value of the textarea will be replaced by the text content stored in the file.
    
Note: The default text within the textarea should be passed using 'value' attribute instead of enclosing the text within the opening and closing tag.

## Usage 
```html
<dnd-textarea
        name = "name"
        value = "some default text"
        placeholder = "placeholder text"
        textarea-style = "style to override default styling of textarea"
    >
</dnd-textarea>
```
## Demo

dnd-textarea demo

PS: I have replaced the default textarea with this webcomponent in two files. If there are any other files that want this update, please let me know in this thread

### CODE REVIEW

File: src/cryptoadvance/specter/templates/includes/dnd-textarea.html [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        constructor() {
+            super();    
+            this.internals = this.attachInternals();
+
+            // Create a shadow root
+            let shadow = this.attachShadow({mode: 'open'});
+            let template = document.getElementById('dnd-textarea');
+            let clone = template.content.cloneNode(true);
+
+            this.$textarea = clone.querySelector("textarea");
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/templates/includes/dnd-textarea.html:53]: Same question regarding the $ sign as here: https://github.com/cryptoadvance/specter-desktop/pull/1752#discussion_r897720045

Comment (b30wulffz) [reply src/cryptoadvance/specter/templates/includes/dnd-textarea.html:53]: I added $ as a custom convention to differentiate between variables containing dom elements and variables containing properties, values and properties. Should I remove them?

Comment (b30wulffz) [reply src/cryptoadvance/specter/templates/includes/dnd-textarea.html:53]: oh sure then, will do it

**Review by moneymanolis — APPROVED** [review]

**Comment by moneymanolis** [comment]
Great work, I just directly fixed the value attribute and added some comments and a link (for more context) since this approach to make a custom element treated like a form control is a new one in the code base.

**Comment by b30wulffz** [comment]
we can merge this now.

File: src/cryptoadvance/specter/templates/wallet/settings/wallet_settings.jinja [hunk 2]
```diff
@@ -187,7 +187,7 @@
 						name="address_labels_data" 
 						id="address_labels_data" 
 						placeholder="{{ _('You can paste and drag & drop here:') }}&#10&#10{{ _('Electrum JSON file, Specter addresses CSV or Specter JSON file') }}"  
-						style="margin-bottom: 10px;">
+						textarea-style="margin-bottom: 10px;">
```
[the remark this answers is not in the document]
Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/wallet/settings/wallet_settings.jinja:190]: Yes, was just for consistency.

**Pull request merged**